File: 백준/Silver/1246. 온라인 판매/온라인 판매.cs
using System;
namespace ConsoleApp1
{class Program
    {
        static void Main(String[] args)
        {
            // n개의 달걀, 잠재 고객은 m명 i번째 고객은 p 이하로 산다고 밝힘..
            // 한 고객에게 두 개 이상의 달걀은 x
            string []tmp = Console.ReadLine().Split(" ");
            int n = int.Parse(tmp[0]);
            int m = int.Parse(tmp[1]);

            int []arr = new int[m];
            for(int i = 0; i < m; i++)
            {
                string egg = Console.ReadLine();
                arr[i] = int.Parse(egg);
            }
            // arr.Sort();
            Array.Sort(arr);
            int x = 0;
            int salePrice = 0;

            for(int i = 0; i < m; i++){
                int restPerson = (m - i);
                if(restPerson > n)
                {
                    restPerson = n;
                }
                if(salePrice < (restPerson * arr[i]))
                {
                    x = arr[i];
                    salePrice = (restPerson * arr[i]);
                }
            }
            Console.WriteLine(x.ToString() + " " + salePrice.ToString());
        }
    }
}

File: 백준/Silver/1246. 온라인 판매/온라인 판매.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main(){
    int n, m;
    cin >> n >> m;

    vector <int> arr(m);
    for(int i = 0; i < m; i++)
        cin >> arr[i];
    
    sort(arr.begin(), arr.end());

    // 최대 판매 가격
    // 팔 수 있는 계란 수 확인
    // 전체 달걀 수 n개 본인 포함 몇 명 남았는지? -> 뒤에서 부터 개수를 세야겠지?
    // 2 7 8 10
    int x = 0, salePrice = 0;
    for(int i = 0; i < m; i++){
        // n개의 달걀.. 
        int restPerson = (m - i); // 본인 포함 팔 수 있는 달걀의 수(뒤에 남은 사람의 수)
        if(restPerson > n)
            restPerson = n; 
        
        if(salePrice < (restPerson * arr[i])){
            x = arr[i];
            salePrice = (restPerson * arr[i]);
        }
    }
    cout << x << " " << salePrice;
}